packit_service/worker/jobs.py

```diff
@@ -107,22 +107,36 @@ def get_config_for_handler_kls(
     :param package_config: we pick the JobConfig from this package_config instance
     :return: JobConfig
     """
+    matching_jobs = []
+    jobs_that_can_be_triggered = []
+
     for job in package_config.jobs:
-        if job.type in MAP_HANDLER_TO_JOB_TYPES[handler_kls] and (
-            (
-                event.db_trigger
-                and event.db_trigger.job_config_trigger_type == job.trigger
-            )
-            or is_trigger_matching_job_config(trigger=event.trigger, job_config=job)
-        ):
-            return job
+        if (
+            event.db_trigger and event.db_trigger.job_config_trigger_type == job.trigger
+        ) or is_trigger_matching_job_config(trigger=event.trigger, job_config=job):
+            jobs_that_can_be_triggered.append(job)
 
-        required_handlers = MAP_REQUIRED_JOB_TO_HANDLERS[job.type]
-        for pos_handler in required_handlers:
+    matching_job_types = MAP_HANDLER_TO_JOB_TYPES[handler_kls]
+    for job in jobs_that_can_be_triggered:
+        if job.type in matching_job_types and job not in matching_jobs:
+            matching_jobs.append(job)
```

Comment:
```suggestion
matching_jobs = {job for job in jobs_that_can_be_triggered if job.type in matching_job_types}
```

Reply:
This won't work b/c job is not hashable (been there, done that)

Reply:
...so Franta would need to put job.type in the set, not the job itself

or fix this in packit first

Reply:
Can we go with the current solution then?

Reply:
fine with me

Reply:
go ahead

Reply:
OK I'll add some more test cases and merge.